records_mover/records/schema/field/constraints/constraints.py

```diff
@@ -108,6 +108,34 @@ def from_data(data: Optional['FieldConstraintsDict'],
         else:
             return RecordsSchemaFieldConstraints(required=required, unique=unique)
 
+    def downcast(self, field_type: 'FieldType') -> 'RecordsSchemaFieldConstraints':
+        from .integer import RecordsSchemaFieldIntegerConstraints
+        from .decimal import RecordsSchemaFieldDecimalConstraints
+        required = self.required
+        unique = self.unique
+        constraints: RecordsSchemaFieldConstraints
+        # TODO: Can I refactor this into a method on these
```

Comment:
Uncompleted punchlist item detected--consider resolving or moving this to your issue tracker